### PR TITLE
Added a global valid method

### PR DESCRIPTION
**Pull request by lucax88x**
Possibility to define an unique validmethod for all the validators

### CODE REVIEW

**Comment by huei90** [comment]
This is useful. :+1: 

and you have your unique documentation format.

**Comment by huei90** [comment]
Better if you can add some test cases, else I can do it.

**Comment by lucax88x** [comment]
Sure, I'll remove my documentation and add some test cases.

**Comment by huei90** [comment]
thank you @lucax88x :beers

**Comment by lucax88x** [comment]
is this unit test sufficient? I couldn't find the "validation" part

**Comment by huei90** [comment]
This is fine, I can add the rest.

one more thing, do you want to add the documentation about how to set global valid method?

**Comment by Nazanin1369** [comment]
Shouldn't the API documentation also be a part of index.html page ?

**Comment by lucax88x** [comment]
I just removed the documentation, I understood you wanted to do it yourself, just tell me where to put it

**Comment by huei90** [comment]
@Nazanin1369 It should be, API documentation html page is our next mission as we discussed in group last week (I think). :smile:

**Comment by huei90** [comment]
Sorry @lucax88x 

You can add below this section

https://github.com/huei90/angular-validation/blob/master/API.md#select-the-validation-method-watch-blur-submit-submit-only-default-as-watch-

Thanks :beers:

**Comment by Nazanin1369** [comment]
I didn't. I ment we have to also include it in our index.html and in the
header menu as "API documentation" to be consistent. Please have that in
mind also.

**Comment by Nazanin1369** [comment]
Got u 😂

**Comment by huei90** [comment]
Oh! I though you mean we need a html structure page for API documentation. For now, we can just add a link in header to direct to API.md

**Comment by lucax88x** [comment]
added documentation @huei90  :)

File: API.md [hunk 1]
```diff
@@ -1,323 +1,354 @@
...
+</script>
+
+<!-- Clean, right ? -->
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+ 
+
+### **Select a global validation method** `watch blur submit submit-only`**
+
+`validationProvider.setValidMethod('submit')`
```
Comment (huei90) [inline API.md]: Sorry, I should say more about this.
I need this section changes only.

We should keep the original documentation style

means we don't need ~~~~~~~~~ style.

**Comment by lucax88x** [comment]
Got it, I didn't notice this change! editor did this...

**Comment by lucax88x** [comment]
fixed it! @huei90

**Comment by huei90** [comment]
Thanks @lucax88x 
Well done. I'm going to merge now :+1: 

Thanks again :beers:

**Comment by huei90** [comment]
@lucax88x I have add one more test case for this feature.

@Nazanin1369 API document added in header.

**Comment by Nazanin1369** [comment]
Great!

**Pull request merged**